libs/core/algorithms/include/hpx/parallel/algorithms/detail/find.hpp

```diff
@@ -256,4 +256,221 @@ namespace hpx { namespace parallel { inline namespace v1 { namespace detail {
+                        ++mid2;
+
+                        for (; local_count != diff;
+                             ++local_count, ++mid, ++mid2)
+                        {
+                            if (!hpx::util::invoke(op,
+                                    hpx::util::invoke(proj1, mid),
+                                    hpx::util::invoke(proj2, *mid2)))
+                            {
+                                break;
+                            }
+                        }
+
+                        if (local_count == diff)
+                        {
+                            tok.cancel(i);
+                        }
+                    }
+                });
+        }
+    };
+
+#if !defined(HPX_COMPUTE_DEVICE_CODE)
+    template <typename ExPolicy>
+    inline constexpr sequential_find_end_t<ExPolicy> sequential_find_end =
+        sequential_find_end_t<ExPolicy>{};
+#else
+    template <typename ExPolicy, typename Iter1, typename Sent1, typename Iter2,
+        typename Sent2, typename Pred, typename Proj1, typename Proj2>
+    HPX_HOST_DEVICE HPX_FORCEINLINE Iter1 sequential_find_end(Iter1 first1,
+        Sent1 last1, Iter2 first2, Sent2 last2, Pred&& op, Proj1&& proj1,
+        Proj2&& proj2)
+    {
+        return sequential_find_end_t<ExPolicy>{}(first1, last1, first2, last2,
+            std::forward<Pred>(op), std::forward<Proj1>(proj1),
+            std::forward<Proj2>(proj2));
+    }
+
+    template <typename ExPolicy, typename Iter1, typename Iter2, typename Token,
+        typename Pred, typename Proj1, typename Proj2>
+    HPX_HOST_DEVICE HPX_FORCEINLINE Iter1 sequential_find_end(Iter1 it,
+        Iter2 first2, std::size_t base_idx, std::size_t part_size,
+        std::size_t diff, Token& tok, Pred&& op, Proj1&& proj1, Proj2&& proj2)
+    {
+        return sequential_find_end_t<ExPolicy>{}(it, first2, base_idx,
+            part_size, diff, tok, std::forward<Pred>(op),
+            std::forward<Proj1>(proj1), std::forward<Proj2>(proj2));
+    }
+#endif
+
+    template <typename ExPolicy>
+    struct sequential_find_first_of_t
+      : hpx::functional::detail::tag_fallback<
+            sequential_find_first_of_t<ExPolicy>>
+    {
+        template <typename InIter1, typename InIter2, typename Pred,
+            typename Proj1, typename Proj2>
+        friend inline constexpr InIter1 tag_fallback_invoke(
+            sequential_find_first_of_t<ExPolicy>, InIter1 first, InIter1 last,
+            InIter2 s_first, InIter2 s_last, Pred&& op, Proj1&& proj1,
+            Proj2&& proj2)
+        {
+            if (first == last)
+                return last;
+
+            for (/* */; first != last; ++first)
+            {
+                for (InIter2 iter = s_first; iter != s_last; ++iter)
+                {
+                    if (hpx::util::invoke(
+                            util::compare_projected<Pred, Proj1, Proj2>(
+                                std::forward<Pred>(op),
```

Comment:
Please use `HPX_FORWARD(Pred, op)` instead of `std::forward<Pred>(op)` wherever possible in implementation files (not tests, etc.).

Reply:
None of `op`, `proj1`, and `proj2` should be forwarded, right? They currently forwarded in a loop?